### DISCUSSION TITLE
Controllo Garage

### DISCUSSION THREAD

**Discussion by nemic** [indomus/forum#798] [GENERALE - Discussioni generiche]
Un saluto a tutto la comunity.
Premetto che sono agli albori e da giorni sto cercando una soluzione, ma per ora non ho trovato risposta. 

**Scenario.**
Ho installato un sonoff per l'apertura e la chiusura del garage e collegato ad alexa. (tutto funziona regolarmente).
Ora il passo successivo è collegare un sensore alla basculante del garage per capire quando è aperto o chiuso.

Domanda:
Una volta installato il sensore (anche esso collegato ad un sonoff), esiste una soluzione per fare in modo che una volta aperta la porta del garage, e passati 60 secondi, se il sensore rileva che è aperta, si possa chiudere automaticamente la porta?

**Comment by riccionese** [comment]
Se hai un sensore che rileva il cambio stato da chiuso ad aperto, puoi sicuramente predisporre una automazione che passati 60 sec AND è ancora aperto ... lo chiuda.
Ma dipende da come funziona la tua basculante. 
La mia non ha un comando "apri" separato da un "chiudi" ma ogni volta che premo l'interruttore (o il telecomando) cambia stato in sequenza: apre-ferma-chiude-ferma-apre-ferma-chiude-ferma. ecc.
Per cui se è "a metà" dipende se s'è fermata in discesa (e quindi si aprirebbe) o in salita (e quindi chiuderebbe)
Io ho risolto con 2 sensori (Aqara Door and Window Sensor) uno per il tutto-aperto ed uno per il tutto-chiuso.
da li in poi non è complicato

**Reply by essegi65** [reply]
Scusa un'alternativa potrebbe essere di far funzionare anche comandando il basculante con uno shelly 2.5 e gestirlo come una tenda con anche la percentuale di apertura o no?

**Reply by nemic** [reply]
Utilizzo sonoff perchè ne avevo alcuni già acquistati. La mia richiesta è dovuta al fatto che voglio evitare che il garage rimanga aperto nel caso mi dimentichi di chiuderlo. Non so' se è fattibile tramuite skill di Alexa, nel caso volevo sapere se usando un hub persoanale come Home assistant fosse possibile programmare la chiusura dopo 60 secondi che è stato verificata l'apertura della porta basculante del garage

**Reply by essegi65** [reply]
Una volta integrato il componente scelto in HA come scritto da @riccionese non hai problemi a creare l'automazione per l'apertura o chiusura dopo un tot di tempo, anche con comandi vocali su alexa e/o google assistant. diverso il caso che dovrai approfondire per stabilire lo stato (se è aperta o chiusa) della basculante.
Certo impostare un hub personale come HA solo per controllare un entità solamente mi sembra eccessivo, anche per la "manutenzione" che richiede, diverso se prendi spunto da questo primo progetto per poi implementare tante altre cose utili come facciamo tutti noi

**Reply by riccionese** [reply]
> Scusa un'alternativa potrebbe essere di far funzionare anche comandando il basculante con uno shelly 2.5 e gestirlo come una tenda con anche la percentuale di apertura o no?

Penso possa funzionare solo se hai comandi separati per apertura e chiusura. 
Nel mio caso, come dicevo sopra, ho un solo pulsante he "cicla" tra apre-feerma-chiude-ferma-apre-ecc ... e non ho idea di come un 2.5 possa essere integrato come fosse una tenda 

P.S.: ho 2 tende da esterno pilotata con shelly 2.5 :-)

**Reply by essegi65** [reply]
@riccionese non mi riferivo al tuo caso :)

**Reply by nemic** [reply]
> Una volta integrato il componente scelto in HA come scritto da @riccionese non hai problemi a creare l'automazione per l'apertura o chiusura dopo un tot di tempo, anche con comandi vocali su alexa e/o google assistant. diverso il caso che dovrai approfondire per stabilire lo stato (se è aperta o chiusa) della basculante. Certo impostare un hub personale come HA solo per controllare un entità solamente mi sembra eccessivo, anche per la "manutenzione" che richiede, diverso se prendi spunto da questo primo progetto per poi implementare tante altre cose utili come facciamo tutti noi

Non ho solo il garage da controllare, ho anche tutte le luci ed il termostato già collegati ad Alexa, nel caso dovessi per forza ricorrere ad un Hub personale per risolvere il mio problema, sposterei anche tutto il resto dei controlli su esso. Per il momento continuo a studiare il mondo delle skill di Alexa e cercare di trovare una soluzione, se proprio non è fattibile penso che mi dovrò per forza compare un Rasp e installarci HA.
